src/custom/components/TransactionConfirmationModal/index.tsx

```diff
@@ -475,53 +188,118 @@ export function ConfirmationPendingContent({
       return WalletType.EOA
     }
   }, [gnosisSafeInfo, isSmartContractWallet])
-
+  const connectionType = getWeb3ReactConnection(connector)
   const walletNameLabel = getWalletNameLabel(walletType)
   const operationMessage = getOperationMessage(operationType, chainId)
   const operationLabel = getOperationLabel(operationType)
   const operationSubmittedMessage = getSubmittedMessage(operationLabel, operationType)
+  const isMetaMask = getIsMetaMask()
+  const isNotMobile = useMediaQuery(MediumAndUp)
+  const isApproveMetaMaskDesktop = operationType === OperationType.APPROVE_TOKEN && isMetaMask && isNotMobile && connectionType !== walletConnectConnection
+
+  // Define Compare Items
+  const compareCapLabel = 'spending cap'
+  const compareTitleLeft = 'Max'
+  const compareTitleRight = 'Use default'
 
   return (
     <Wrapper>
       <UpperSection>
         <CloseIconWrapper onClick={onDismiss} />
-
-        <WalletIcon>{getStatusIcon(connector, walletDetails, 46)}</WalletIcon>
-
-        <Text fontWeight={500} fontSize={16} textAlign="center">
-          {pendingText}
-        </Text>
+        <WalletIcon>{getStatusIcon(connector, walletDetails, 56)}</WalletIcon>
+        <span>{pendingText}</span>
       </UpperSection>
 
-      <LowerSection>
-        <h3>
-          <span>{operationMessage} </span>
-          <span>
-            <Trans>Follow these steps:</Trans>
-          </span>
-        </h3>
-
-        <StepsWrapper>
-          <div>
-            <StepsIconWrapper>
-              <UserCheck />
-            </StepsIconWrapper>
-            <p>
-              <Trans>
-                Sign the {operationLabel} with your {walletNameLabel}{' '}
-                {account && <span>({ensName || shortenAddress(account)})</span>}
-              </Trans>
-            </p>
-          </div>
-          <hr />
-          <div>
-            <StepsIconWrapper>
-              <CheckCircle />
-            </StepsIconWrapper>
-            <p>{operationSubmittedMessage}</p>
-          </div>
-        </StepsWrapper>
-      </LowerSection>
+      {/* Only shown for APPROVE_TOKEN operation for MetaMask Desktop */}
+      {isApproveMetaMaskDesktop && (
```

Comment:
Would be great to split the component:
`{isApproveMetaMaskDesktop && <ApproveMetaMaskDesktop/>}`